hub/saml-engine/src/integration-test/java/uk/gov/ida/integrationtest/hub/samlengine/apprule/IdpAuthnResponseTranslatorResourceWithRedisTest.java

```diff
@@ -0,0 +1,176 @@
+package uk.gov.ida.integrationtest.hub.samlengine.apprule;
+
+import helpers.JerseyClientConfigurationBuilder;
+import io.dropwizard.client.JerseyClientBuilder;
+import io.dropwizard.client.JerseyClientConfiguration;
+import io.dropwizard.testing.ConfigOverride;
+import io.dropwizard.util.Duration;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import uk.gov.ida.common.ErrorStatusDto;
+import uk.gov.ida.common.ExceptionType;
+import uk.gov.ida.hub.samlengine.Urls;
+import uk.gov.ida.hub.samlengine.contracts.SamlAuthnResponseTranslatorDto;
+import uk.gov.ida.integrationtest.hub.samlengine.apprule.support.ConfigStubRule;
+import uk.gov.ida.integrationtest.hub.samlengine.apprule.support.RedisTestRule;
+import uk.gov.ida.integrationtest.hub.samlengine.apprule.support.SamlEngineAppRule;
+import uk.gov.ida.integrationtest.hub.samlengine.builders.AuthnResponseFactory;
+import uk.gov.ida.shared.utils.datetime.DateTimeFreezer;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.ida.integrationtest.hub.samlengine.builders.SamlAuthnResponseTranslatorDtoBuilder.aSamlAuthnResponseTranslatorDto;
+import static uk.gov.ida.saml.core.test.TestEntityIds.STUB_IDP_ONE;
+import static uk.gov.ida.saml.core.test.TestEntityIds.STUB_IDP_THREE;
+import static uk.gov.ida.saml.core.test.TestEntityIds.STUB_IDP_TWO;
+import static uk.gov.ida.saml.core.test.TestEntityIds.TEST_RP_MS;
+
+public class IdpAuthnResponseTranslatorResourceWithRedisTest {
+
+    private static Client client;
+
+    private final String IDP_RESPONSE_ENDPOINT = "http://localhost" + Urls.FrontendUrls.SAML2_SSO_RESPONSE_ENDPOINT;
+    private final AuthnResponseFactory authnResponseFactory = new AuthnResponseFactory();
+
+    private static RedisTestRule redis = new RedisTestRule(6379);
```

Comment:
Can we shut Redis server down at the end of all tests? I got the following error message after running pre-commit.sh twice.

```java.lang.RuntimeException: Can't start redis server. Check logs for details.```